### PR TITLE
Add Topic#persistence_regions

### LINKED ISSUE #3389: Support message_storage_policy in hand-written UI
Topic#message_storage_policy was added in #2237, add this to the hand-written UI.

### PR DESCRIPTION
**Pull request by blowmage**
This PR adds the `Topic#persistence_regions` method to the public API. It calls `MessageStoragePolicy#allowed_persistence_regions` from the low-level API.

[closes #3389]

### CODE REVIEW

File: google-cloud-pubsub/lib/google/cloud/pubsub/topic.rb [hunk 1]
```diff
@@ -182,6 +182,36 @@ def kms_key= new_kms_key_name
...
+        # The list of GCP region IDs where messages that are published to the
+        # topic may be persisted in storage.
+        #
+        # Messages published by publishers running in non-allowed GCP regions
+        # (or running outside of GCP altogether) will be routed for storage in
+        # one of the allowed regions. An empty list indicates a misconfiguration
+        # at the project or organization level, which will result in all publish
+        # operations failing.
+        #
+        # Makes an API call to retrieve the labels values when called on a
```
Comment (quartzmo) [inline google-cloud-pubsub/lib/google/cloud/pubsub/topic.rb]: fix "labels" c/p

File: google-cloud-pubsub/lib/google/cloud/pubsub/topic.rb [hunk 2]
```diff
@@ -182,6 +182,36 @@ def kms_key= new_kms_key_name
...
+        #
+        #   pubsub = Google::Cloud::PubSub.new
+        #
+        #   topic = pubsub.topic "my-topic"
+        #
+        #   topic.persistence_regions #=> ["us-central1", "us-central2"]
+        #
+        def persistence_regions
+          ensure_grpc!
+          return [] if @grpc.message_storage_policy.nil?
```
Comment (quartzmo) [inline google-cloud-pubsub/lib/google/cloud/pubsub/topic.rb]: Looks to me like the truthy condition here needs a unit test.

File: google-cloud-pubsub/lib/google/cloud/pubsub/topic.rb (hunk 1 above)
Comment (blowmage) [reply google-cloud-pubsub/lib/google/cloud/pubsub/topic.rb]: Fixed.

File: google-cloud-pubsub/lib/google/cloud/pubsub/topic.rb (hunk 2 above)
Comment (blowmage) [reply google-cloud-pubsub/lib/google/cloud/pubsub/topic.rb]: Added.

**Pull request merged**